File: cpp/02-gtest-ut/src/FizzBuzz2.cpp
#include "FizzBuzz2.hpp"
#include <iostream>

void FizzBuzz2::fizzBuzz0(int n) {
    for (int i = 1; i <= n; i++) {
        std::string v = fizzBuzz(i);
        std::cout << v << std::endl;
    }
}

std::string FizzBuzz2::fizzBuzz(int i) {
    bool n是3的倍数 = i % 3 == 0;
    bool n是5的倍数 = i % 5 == 0;
    if (n是3的倍数 && n是5的倍数) {
        return "FizzBuzz";
    } else if (n是3的倍数) {
        return "Fizz";
    } else if (n是5的倍数) {
        return "Buzz";
    } else {
        return std::to_string(i);
    }
}

File: javascript/02-mocha/src/fizzbuzz2.js
function fizzBuzz(n) {
  const isMultipleOf3 = n % 3 === 0;
  const isMultipleOf5 = n % 5 === 0;

  if (isMultipleOf3 && isMultipleOf5) {
    return 'FizzBuzz';
  }
  if (isMultipleOf3) {
    return 'Fizz';
  }
  if (isMultipleOf5) {
    return 'Buzz';
  }
  return String(n);
}

module.exports = { fizzBuzz };
